File: sdks/java/lib/src/main/java/models/shared/DevicePinner.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * DevicePinner
 * Device pins allow enterprises to control what devices can
 * use native Box applications.
**/
public class DevicePinner {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public DevicePinner withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("owned_by")
    public UserMini ownedBy;
    public DevicePinner withOwnedBy(UserMini ownedBy) {
        this.ownedBy = ownedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("product_name")
    public String productName;
    public DevicePinner withProductName(String productName) {
        this.productName = productName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public DevicePinnerTypeEnum type;
    public DevicePinner withType(DevicePinnerTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: python-sdk/src/boxsdk/models/shared/devicepinner.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import user_mini as shared_user_mini

class DevicePinnerTypeEnum(str, Enum):
    DEVICE_PINNER = "device_pinner"


@dataclass_json
@dataclasses.dataclass
class DevicePinner:
    r"""DevicePinner
    Device pins allow enterprises to control what devices can
    use native Box applications.
    """
    
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    owned_by: Optional[shared_user_mini.UserMini] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('owned_by') }})
    product_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('product_name') }})
    type: Optional[DevicePinnerTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
